config/config.go

```diff
@@ -2,90 +2,100 @@ package config
 
 import (
 	"fmt"
-	"math/big"
 	"github.com/spf13/viper"
 	"github.com/xlab-si/emmy/dlog"
+	"math/big"
 )
 
+// init Loads the default config file
 func init() {
 	viper.AddConfigPath("$GOPATH/src/github.com/xlab-si/emmy/config")
+	LoadConfig("defaults", "yml")
 }
 
-// Type can be "yml", "json" ...
-func LoadConfig(configName string, ctype string) {
+// LoadConfig reads in the config file with configName being the name of the file (without suffix)
+// and configType being "yml" or "json".
+func LoadConfig(configName string, configType string) {
 	viper.SetConfigName(configName)
-	viper.SetConfigType(ctype)
+	viper.SetConfigType(configType)
 
 	err := viper.ReadInConfig()
 	if err != nil {
 		panic(fmt.Errorf("Cannot read configuration file: %s\n", err))
 	}
 }
 
-func LoadKeyDirFromConfig() (string) {
-	LoadConfig("cli", "yml")
+// LoadServerPort returns the port where emmy server will be listening.
+func LoadServerPort() int {
+	return viper.GetInt("port")
+}
+
+// LoadServerEndpoint returns the endpoint of the emmy server where clients will be contacting it.
+func LoadServerEndpoint() string {
+	ip := viper.GetString("ip")
+	port := LoadServerPort()
+	return fmt.Sprintf("%v:%v", ip, port)
+}
+
+// LoadTimeout returns the specified number of seconds that clients wait before giving up
+// on connection to emmy server
+func LoadTimeout() float64 {
+	return viper.GetFloat64("timeout")
+}
+
+func LoadKeyDirFromConfig() string {
 	key_path := viper.GetString("key_folder")
 	return key_path
 }
 
-func LoadTestKeyDirFromConfig() (string) {
-	LoadConfig("test", "yml")
+func LoadTestKeyDirFromConfig() string {
 	key_path := viper.GetString("key_folder")
 	return key_path
 }
 
-func LoadPseudonymsysDLog() *dlog.ZpDLog {
-	LoadConfig("dlogs", "json")
-	dlogMap := viper.GetStringMap("pseudonymsys")
-	p, _ := new(big.Int).SetString(dlogMap["P"].(string), 10)
-	g, _ := new(big.Int).SetString(dlogMap["G"].(string), 10)
-	q, _ := new(big.Int).SetString(dlogMap["Q"].(string), 10)
+func LoadDLog(scheme string) *dlog.ZpDLog {
+	dlogMap := viper.GetStringMap(scheme)
+	p, _ := new(big.Int).SetString(dlogMap["p"].(string), 10)
```

Comment:
Actually, if the config file is not found, the programs (emmy server, emmy clients) panic immediately. So the `dlog.p` won't be able to complain, as the program will exit prior to that. This is due to following section of code in `config.go`'s `LoadConfig` function (L18-L26), which is called by `init`:
```go
err := viper.ReadInConfig()
if err != nil {
	panic(fmt.Errorf("Cannot read configuration file: %s\n", err))
}
```